api/v1beta2/moduletemplate_types.go

```diff
@@ -138,11 +138,34 @@ type ModuleTemplateSpec struct {
 
 	// Info contains metadata about the module.
 	// +optional
-	Info ModuleInfo `json:"info,omitempty"`
+	Info *ModuleInfo `json:"info,omitempty"`
```

Comment:
Why is this change necessary?

Reply:
use pointer will make the actual code create this `ModuleInfo` a bit easier, with `nil` providing null field, and not necessarily creating an empty `ModuleInfo` struct.